### PR TITLE
Calculate number of dots, allow to change per breakpoint

### PR DESCRIPTION
**Pull request by zackphilipps**
This is still WIP, and is more of a proof of concept. The main thing on my to-do list right now is to apply these changes to the actual `CarouselDots` component as well. But, I'm not really sure how to do that at the moment since the number of dots depends on `Carousel` props and they are two separate components. Still fairly new to React. 😅 

Anyway, let me know what you think!

- [ ] `package.json:version` has been updated with `npm version patch` (or `major/minor` as appropriate)
- [ ] `@brainhubeu/react-carousel` version has been updated in `docs-www/package.json` to the same which is in `package.json:version`

### CODE REVIEW

File: src/components/Carousel.js [hunk 1]
```diff
@@ -280,16 +282,39 @@ export default class Carousel extends Component {
...
+    if (!numDots) {
+      numDots = 1;
+    }
+
+    if (numSlides % slidesPerPage || slidesPerPage === 1) {
+      numDots++;
+    }
+
+    if (numDots !== this.state.numDots) {
+      this.setState({ value: 0 });
```
Comment (zackphilipps) [inline src/components/Carousel.js]: actually, this is supposed to move the slider to 0 when a configured breakpoint is hit, and this isn't working. couldn't get `this.changeSlide(0)` to work with the unit tests though.

**Comment by piotr-s-brainhub** [comment]
@zackphilipps 

could you resolve conflicts?

**Comment by piotr-s-brainhub** [comment]
@zackphilipps 

Thanks for this PR.
Could you resolve the conflicts?

**Comment by piotr-s-brainhub** [comment]
@zackphilipps ?

**Comment by zackphilipps** [comment]
@piotr-s-brainhub this probably requires more testing than I have time for right now. As I recall, after opening this PR, I discovered more issues with dots / breakpoints and opted to use react-slick instead.

**Comment by piotr-s-brainhub** [comment]
@zackphilipps 

thanks for the reply

**Comment by Lukasz-pluszczewski** [comment]
This PR seems to be dead, and also, the issue with different number of dots than expected by some developers, namely the fact that number of dots is always the number of slides passed to the carousel, not the number of pages, has been discussed in a different PR. My conclusion is that current behaviour is much simpler to maintain, and dots can easily be implemented outside carousel by developer.

**Pull request closed without merging**